src/dialer/index.js

```diff
@@ -190,6 +202,48 @@ class Dialer {
     log('token %d released', token)
     this.tokens.push(token)
   }
+
+  /**
+   * Resolve multiaddr recursively.
+   *
+   * @param {Multiaddr} ma
+   * @returns {Promise<Array<Multiaddr>>}
+   */
+  async _resolve (ma) {
+    // TODO: recursive logic should live in multiaddr once dns4/dns6 support is in place
+    const resolvableProto = ma.protos().find((p) => p.resolvable)
```

Comment:
I think we should simplify this for now and **only** resolve if `ma.protoNames().includes('dnsaddr')`. The checks below don't account for `dns` addresses. If new resolvable addresses are added, we may need to account for them further, so I would just simplify this for now.

Reply:
yes, I agree!